src/transforms/filter.js

```diff
@@ -190,85 +187,71 @@ function getFilterFunc(opts, d2c) {
-                    return cv > coercedValue[0] && cv < coercedValue[1];
-                };
-            }
-            else if(coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv > coercedValue[0] && cv <= coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv >= coercedValue[0] && cv < coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv >= coercedValue[0] && cv <= coercedValue[1];
-                };
-            }
-
-            break;
-
-        case 'notwithin':
-
-            if(coercedStrict[0] && coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv < coercedValue[0] || cv > coercedValue[1];
-                };
-            }
-            else if(coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv < coercedValue[0] || cv >= coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv <= coercedValue[0] || cv > coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv <= coercedValue[0] || cv >= coercedValue[1];
-                };
-            }
-
-            break;
-
-        case 'in':
+            return function(v) { return d2c(v) > coercedValue; };
+
+        case '>=':
+            return function(v) { return d2c(v) >= coercedValue; };
+
+        case '[]':
+            return function(v) {
+                var cv = d2c(v);
+                return cv >= coercedValue[0] && cv <= coercedValue[1];
+            };
+
+        case '()':
+            return function(v) {
+                var cv = d2c(v);
+                return cv > coercedValue[0] && cv < coercedValue[1];
+            };
+
+        case '[)':
+            return function(v) {
+                var cv = d2c(v);
+                return cv >= coercedValue[0] && cv < coercedValue[1];
+            };
+
+        case '(]':
+            return function(v) {
+                var cv = d2c(v);
+                return cv > coercedValue[0] && cv <= coercedValue[1];
+            };
+
+        case '][':
+            return function(v) {
+                var cv = d2c(v);
+                return cv < coercedValue[0] || cv > coercedValue[1];
```

Comment:
@etpinard has already changed it per my suggestion (thanks!) so not to beat a dead horse but I find it much more persuasive to think of these as a combination of two independent operators (`]` always means `<=`) than to try to relate them to the corresponding "within" forms. Likewise, think about histograms: if you have a bin with an interval like `[)`, then to meet it exactly the next bin must start with `[`, hence `[` is the complement of `)`